src/NUnitFramework/tests/Assertions/AssertThrowsAsyncTests.cs

```diff
@@ -0,0 +1,214 @@
+﻿using System;
+using System.Threading.Tasks;
+using NUnit.TestUtilities;
+
+namespace NUnit.Framework.Tests.Assertions
+{
+    [TestFixture]
+    public class AssertThrowsAsyncTests
+    {
```

Comment:
> These would break some of our builds and so needs to be protected by an #if statement.
> 
> Note that our general practice is to include framework files in all the framework builds but #if them out where not wanted. That facilitates searching the projects for all the #if statements.

OK, I thought it would be cleaner to include it only in the 4.5 test project, but I can change it.
BTW, I didn't include these tests in nunit.framework.tests-4.0, because the tests would be much harder to write without async/await support. Is this an issue? An option would be to reference `Microsoft.Bcl.Async` to support async/await in 4.0 (we can't do that in nunit.framework, but I don't think it would be a problem for a test project).

> This is the first point where I noticed it but it applies to other files as well.

I did it for Assert.Exceptions.Async.cs as well, which is only included in nunit.framework-4.0 and 4.5. I'll include it in the other projects as well.